### ISSUE TITLE
make the corrected energy the default in CaloCluster 

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmssw#22079]
The CaloCluster corrected energy was introduced in https://github.com/cms-sw/cmssw/pull/686
Now that e.g. in #22071 we start to spread the use correctedEnergy, it seems like the choice was not the best for general utility.

From just a general perspective, I think that "energy()" is supposed to correspond to the best known value and the user should not guess that a better value is "hidden" in a correctedEnergy accessor.

I propose to make what's returned by correctedEnergy to be the default and move the uncalibrated value to rawEnergy as we have it in SuperCluster.

@lgray 
please remind me the rationale (if any) for the choice.
Looking at the notes in  #686, it seems like the choice was driven by technical convenience (to introduce zero changes).

@lgray @bachtis @arizzi @gpetruc 
please comment.

**Comment by slava77** [comment]
assign reconstruction

**Comment by lgray** [comment]
@slava77 Yes, this was originally done to make sure that there were no changes at all to physics quantities.

**Comment by bendavid** [comment]
I would be concerned that downstream analysis code for pog/dpg workflows might assume energy() is the raw energy and have unexpected changes introduced by this.

**Comment by slava77** [comment]
On 2/1/18 1:26 PM, Josh Bendavid wrote:
> I would be concerned that downstream analysis code for pog/dpg workflows
> might assume energy() is the raw energy and have unexpected changes
> introduced by this.
> 

cases available in CMSSW should be relatively easy to spot.

What's happening on the other side, btw: is every place that is supposed
to use a corrected energy actually picking it up?
